aiidalab_widgets_base/structures.py

```diff
@@ -1,88 +1,138 @@
 from __future__ import print_function
 from __future__ import absolute_import
 
+import os
 import ase.io
 import ipywidgets as ipw
 from fileupload import FileUploadWidget
 import tempfile
 import nglview
 from six.moves import zip
+from traitlets import Bool
 
 
 class StructureUploadWidget(ipw.VBox):
-
+    has_structure = Bool(False)
     DATA_FORMATS = ('StructureData', 'CifData')
-
-    def __init__(self, text="Upload Structure", node_class=None, **kwargs):
+    def __init__(self, text="Upload Structure", storable=True, node_class=None, examples=[], **kwargs):
         """ Upload a structure and store it in AiiDA database.
 
         :param text: Text to display before upload button
         :type text: str
+        :param storable: Whether to provide Store button (together with Store format)
+        :type storable: bool
         :param node_class: AiiDA node class for storing the structure.
             Possible values: 'StructureData', 'CifData' or None (let the user decide).
             Note: If your workflows require a specific node class, better fix it here.
+        :param examples: list of paths to the example structures
+        :type examples: list
 
         """
-
         self.file_upload = FileUploadWidget(text)
+        self.select_example = ipw.Dropdown(
+            options=self.get_example_structures(examples),
+            description='Or choose from examples:',
+            style={'description_width': '160px'},
+        )
+        supported_formats = ipw.HTML("""All supported structure formats are listed
+        <a href="https://wiki.fysik.dtu.dk/ase/_modules/ase/io/formats.html" target="_blank">here</a>""")
+
         self.viewer = nglview.NGLWidget()
         self.btn_store = ipw.Button(
             description='Store in AiiDA', disabled=True)
         self.structure_description = ipw.Text(
             placeholder="Description (optional)")
 
         self.structure_ase = None
-        self.structure_node = None
+        self._structure_node = None
```

Comment:
why is this no longer public?
I would say this variable should actually be the traitlet, no?
Then there is no need for has_structure

Reply:
see https://github.com/aiidalab/aiidalab-widgets-base/issues/1 on how to achieve this

Reply:
scratch that - I think your approach is fine.